app/components/data-download.tsx

```diff
@@ -134,62 +137,44 @@ export const AllAndVisibleDataDownloadMenu = memo(
     title: string;
     chartConfig: ChartConfig;
   }) => {
-    const allPrepared = usePreparedAllData({ dataSetIri });
-    const visiblePrepared = usePreparedVisibleData({ dataSetIri, chartConfig });
-    const optionsToRender = useMemo(
-      () =>
-        OPTIONS_TO_RENDER.map((d) => ({
-          ...d,
-          columnKeys:
-            d.extent === "all"
-              ? allPrepared.columnKeys
-              : visiblePrepared.columnKeys,
-          data: d.extent === "all" ? allPrepared.data : visiblePrepared.data,
-        })),
-      [allPrepared, visiblePrepared]
-    );
+    const preparedAllData = usePreparedAllData({ dataSetIri });
+    const preparedVisibleData = usePreparedVisibleData({
+      dataSetIri,
+      chartConfig,
+    });
```

Comment:
Does this mean that we download all the data, and all the visible data in every case ? For performance reasons, maybe we should only download the data when the user has opened the menu ?

Reply:
I refactored the code to actually download the data when user clicks a button, thanks for pointing this out :)